internal/controller/ocirepository_controller.go

```diff
@@ -742,35 +751,23 @@ func (r *OCIRepositoryReconciler) verifySignature(ctx context.Context, obj *ociv
 			notation.WithLogger(ctrl.LoggerFrom(ctx)),
 		}
 
-		for k, data := range pubSecret.Data {
-			if strings.HasSuffix(k, ".crt") || strings.HasSuffix(k, ".pem") {
-				verifier, err := notation.NewNotationVerifier(append(
-					defaultNotationOciOpts,
-					notation.WithRootCertificate(data))...)
-				if err != nil {
-					return soci.VerificationResultFailed, err
-				}
-
-				result, err := verifier.Verify(ctxTimeout, ref)
-				if err != nil {
-					if errors.As(err, &notationgo.ErrorSignatureRetrievalFailed{}) {
-						continue
-					}
-					return result, err
-				}
+		verifier, err := notation.NewNotationVerifier(append(
+			defaultNotationOciOpts,
+			notation.WithRootCertificate(certs))...)
+		if err != nil {
+			return soci.VerificationResultFailed, err
+		}
 
-				if result != soci.VerificationResultFailed {
-					verifiedResult = result
-					break
-				}
-			}
+		result, err := verifier.Verify(ctxTimeout, ref)
+		if err != nil {
+			return result, err
 		}
 
-		if verifiedResult == soci.VerificationResultFailed {
+		if result == soci.VerificationResultFailed {
 			return soci.VerificationResultFailed, fmt.Errorf("no matching signatures were found for '%s'", ref)
 		}
 
-		return verifiedResult, nil
+		return result, nil
 	}
 
 	return soci.VerificationResultSuccess, nil
```

Comment:
I just noticed this. This should never happen but we want the default to be `provider not supported`

Reply:
I copied the same default found in `helmchart_controller.go` [ea2d4ca](https://github.com/fluxcd/source-controller/pull/1075/commits/ea2d4ca819716008b05a4b6d35708828879f72ae).